### ISSUE TITLE
Mypy complains about subclassing StacIO or DefaultStacIO

### ISSUE THREAD

**Issue by l0b0** [stac-utils/pystac#582]
How to reproduce:

1. Create test.py with the following contents:

   ```python
   from pystac import StacIO
   
   
   class S3StacIO(StacIO):
       pass
   ```
2. Run `mypy --strict test.py`

Expected result: This should pass type checks.

Actual result:

```text
test.py:4: error: Class cannot subclass "StacIO" (has type "Any")  [misc]
Found 1 error in 1 file (checked 1 source file)
```

**Comment by duckontheweb** [comment]
I believe that this will be fixed by #579, but I'll try to get out a patch release soon so you can verify that.

**Comment by duckontheweb** [comment]
In PySTAC v1.1.0 I get 2 errors running this:

```console
$ mypy --strict test.py
test.py:1: error: Module "pystac" does not explicitly export attribute "StacIO"; implicit reexport disabled
test.py:3: error: Class cannot subclass "StacIO" (has type "Any")
Found 2 errors in 1 file (checked 1 source file)
```

It looks like both errors are fixed by changing the import:

```python
# test.py
from pystac.stac_io import StacIO

class S3StacIO(StacIO):
    pass
```

```console
$ mypy --strict test.py
Success: no issues found in 1 source file
```

Looks like we need to fix those implicit re-exports :)

**Comment by duckontheweb** [comment]
Verified that this is fixed by #591
